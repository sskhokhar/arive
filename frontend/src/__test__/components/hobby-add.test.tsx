import React from 'react';
import { mount } from 'enzyme';
import HobbyItem from '../../components/hobby/hobby';

describe('<HobbyItem />', () => {
  let component: any;
  let clickFn = jest.fn();
  beforeAll(() => {
    component = mount(<HobbyItem onDelete={clickFn} />);
  });
  it('should render the item', () => {
    expect(component.html()).toMatchSnapshot();
  });
  it('should calls the onClick prop function when clicked', () => {
    expect(clickFn).not.toHaveBeenCalled();
    component.find('button').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});
