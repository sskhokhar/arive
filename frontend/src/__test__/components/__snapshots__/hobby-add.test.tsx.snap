// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"<div class=\\"hobby-wrapper\\"><span>Passion Level: </span><span>Name: </span><span>Since: </span><span><button class=\\"btn btn-warn\\">Delete</button></span></div>"`;

exports[`<HobbyItem /> should render the item 1`] = `"<div class=\\"hobby-wrapper\\"><span>Passion Level: </span><span>Name: </span><span>Since: </span><span><button class=\\"btn btn-warn\\">Delete</button></span></div>"`;

exports[`<HobbyItem /> should render the props 1`] = `"<div class=\\"hobby-wrapper\\"><span>Passion Level: </span><span>Name: </span><span>Since: </span><span><button class=\\"btn btn-warn\\">Delete</button></span></div>"`;
